featuretools/tests/synthesis/test_dfs_method.py

```diff
@@ -53,6 +56,44 @@ def datetime_es():
     return datetime_es
 
 
+def test_dfs_empty_features():
+    error_text = 'No features can be generated from the specified primitives. Please make sure the primitives you are using are compatible with the variable types in your data.'
+    with pytest.raises(AssertionError, match=error_text), patch.object(DeepFeatureSynthesis,
+                                                                       'build_features', return_value=[]):
+        teams = pd.DataFrame({
+            'id': range(3),
+            'name': ['Breakers', 'Spirit', 'Thorns']
+        })
+        games = pd.DataFrame({
+            'id': range(5),
+            'home_team_id': [2, 2, 1, 0, 1],
+            'away_team_id': [1, 0, 2, 1, 0],
+            'home_team_score': [3, 0, 1, 0, 4],
+            'away_team_score': [2, 1, 2, 0, 0]
+        })
+        dataframes = {'teams': (teams, 'id', None, {'name': 'natural_language'}), 'games': (games, 'id')}
+        relationships = [('teams', 'id', 'games', 'home_team_id')]
+        dfs(dataframes, relationships, target_dataframe_name="teams", features_only=False)
+
+
+def test_dfs_empty_features_features_only():
```

Comment:
instead of a second test, what if the first test had a structure like this

```
with patch(DFSObject):
    test features_only=True returns empty list
    with pytest.raises(AssertionError):
        test features_only=False raises expected error message
```

that way we'd test if we can return an empty list of features in the features_only case

Reply:
combined test cases for either feature_only or not